torchvision/prototype/transforms/functional/utils.py

```diff
@@ -0,0 +1,72 @@
+import functools
+from typing import Any, Type, Optional, Callable
+
+import PIL.Image
+import torch
+import torch.overrides
+from torchvision.prototype import features
+from torchvision.prototype.utils._internal import sequence_to_str
+
+
+def is_supported(obj: Any, *types: Type) -> bool:
+    return (obj if isinstance(obj, type) else type(obj)) in types
+
+
+class Dispatcher:
+    FEATURE_SPECIFIC_PARAM = object()
+
+    def __init__(self, dispatch_fn):
+        self._dispatch_fn = dispatch_fn
+        self._support = set()
+        self._pil_kernel: Optional[Callable] = None
+
+    def supports(self, obj: Any) -> bool:
+        return is_supported(obj, *self._support)
+
+    def implements(self, feature_type, *, feature_specific_params=(), pil_kernel=None):
+        if pil_kernel is not None:
+            if not issubclass(feature_type, features.Image):
+                raise TypeError("PIL kernel can only be registered for images")
+
+            self._pil_kernel = pil_kernel
+
+        def outer_wrapper(implement_fn):
+            feature_type._KERNELS[self._dispatch_fn] = implement_fn
+            self._support.add(feature_type)
+
+            @functools.wraps(implement_fn)
+            def inner_wrapper(*args, **kwargs) -> Any:
+                missing = [
+                    param
+                    for param in feature_specific_params
+                    if kwargs.get(param, self.FEATURE_SPECIFIC_PARAM) is self.FEATURE_SPECIFIC_PARAM
+                ]
+                if missing:
+                    raise TypeError(
+                        f"{implement_fn.__name__}() missing {len(missing)} required keyword-only arguments: "
+                        f"{sequence_to_str(missing, separate_last='and ')}"
+                    )
```

Comment:
There are cases where only one kernel needs extra information. For example, `cutmix_image` needs the `box` argument, while `cutmix_one_hot_label` does not. In order for the dispatcher to also work for labels if `box` is not supplied, we use a sentinel as default argument and emulate the error message here.